README.md

```diff
@@ -188,4 +186,54 @@ final class HAL9000ViewController: UIViewController {
         podBayInteraction.didTapMainButton()
     }
 }
+```
+
+## 🛠 Assembler
+
+> Assembler
+> (*noun*)
+> 
+> Assembler is assembling dependencies and instantiating one component.
+
+### Definitions
+
+### TL;DR
+
+Its purpose is to assemble dependencies for one class or struct. 
+
+#### Assembler:
+- a struct,
+- as only one method called `assemble()`. 
+- can call ONLY ONE initializer directly,
+- will call other Assemblers to instantiate dependencies.
+
+### Example
+
+```swift
+protocol PodBayDoorsInteractionAssembling {
+	func assemble() -> PodBayDoorsInteracting
+}
+
+struct PodBayDoorsInteractionAssembler {
+    func assemble() -> PodBayDoorsInteracting {
+        return PodBayDoorsInteraction()
+    }
+}
+
+protocol DiscoveryOneAssembling {
+    func assemble() -> UIViewController
+}
+
+struct DiscoveryOneAssembler: DiscoveryOneAssembling {
```

Comment:
definitely prefer the lifetime version using a `struct` than the non-lifetime `enum` version 👍